### DISCUSSION TITLE
Symfony 5.3: Simply disable the cache

### DISCUSSION THREAD

**Discussion by kimjohans** [symfony/symfony#43551] [Q&A]
Hello all,

there are a lot of details about configuring the cache (https://symfony.com/doc/current/cache.html) but the most important and 1st question is nowhere answered:

How can i simply turn of the cache completly ?

Thanks

**Comment by alessandro-podo** [comment]
i think thats not possible, because die DI needs it. 
Why do you want to disable the Cache? 

Edit: 
Thats should be the Answer https://stackoverflow.com/questions/65863552/how-to-disable-cache-in-symfony-5-2

**Reply by kimjohans** [reply]
(a) because my app is encoded and the cache is showing decoded details. not the meaning of encoding.
(b) because i am currently in dev-mode and there the cache is useless

I think, developers should be able to deactivate caching completely (like sessions)

**Reply by kimjohans** [reply]
Yes, i saw that stackoverflow thread during my searches, but the idea is not working.

even with
`use Symfony\Component\Cache\Adapter\NullAdapter;`
and
`$framework->cache()`
`	->app(NullAdapter) // "cache.adapter.null" is also not working`
`	->system(NullAdapter);`
the cache is being written.

**Reply by alessandro-podo** [reply]
sry then I can not help. Hopefully, someone with more knowledge will get in touch.

**Reply by kimjohans** [reply]
Thanks for your attention anyway !

**Reply by stof** [reply]
Well, the DIC cache is not stored using the cache component, and so won't be impacted by the configuration using a NullAdapter.

And the DIC cache cannot be disabled. The container building process generates PHP code for the compiled container, and that code needs to be written somewhere to be used (btw, even in dev, you benefit from it, as you will often do several HTTP requests without changing the configuration, where you will benefit from reusing the compiled container)
